workspaces/rstudio/Dockerfile

```diff
@@ -0,0 +1,120 @@
+
+# PrairieLearn RStudio workspace template
+# 20210831 Eric Huber
+
+FROM rocker/rstudio
+
+# On PL, we need to use 1001:1001 for the user account.
+USER 0
+RUN export OLD_UID=$(id -u rstudio) && \
+    export OLD_GID=$(id -g rstudio) && \
+    export NEW_UID=1001 && \
+    export NEW_GID=1001 && \
+    groupmod -g 1001 rstudio && \
+    usermod -u 1001 -g 1001 rstudio && \
+    find /home -user $OLD_UID -execdir chown -h $NEW_UID {} + && \
+    find /home -group $OLD_GID -execdir chgrp -h $NEW_GID {} + && \
+    unset OLD_UID OLD_GID NEW_UID NEW_GID
+
+USER 0
+RUN apt-get update && \
+    apt-get install -y --no-install-recommends \
+    # Proxy workaround for PrairieLearn:
+    nginx \
+    gettext \
+    fonts-dejavu \
+    # Things for our sanity in the instructor terminal:
+    less htop vim nano silversearcher-ag zip unzip git cmake curl wget && \
+    # Cleanup:
+    apt-get clean && \
+    rm -rf /var/lib/apt/lists/* && \
+    find /tmp -not -path /tmp -delete
+
+# Make sure we'll be able to run as user 1001. This is hacky.
+USER 0
+RUN mkdir -p /var/run/s6 && \
+    chown -R 1001:1001 /var/run/s6 && \
+    chown -R 1001:1001 /usr/local/lib/R && \
```

Comment:
There is an old precedent here of making the (effectively single-user) user (of the multi-user) be part of (either one of) `staff` or `adm` (and I never remember which).  

If it is just one user, setting uid=gid is fine too, of course.  The main point, and it is important, is to allow user to install packages as non-root, and this ensures it.

Reply:
Ah, I figured if anything more needed to be installed in the workspace, then the admins had better roll a new image (otherwise, it'll be hundreds of students reinstalling ephemeral packages every time they reload the workspace).